Upstream PR: https://github.com/google/s2geometry/pull/161

From 132f14fdd15fff532b5ad78ce1c5fedd242164d3 Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Sat, 5 Dec 2020 14:24:27 +0100
Subject: [PATCH] CMake: add version information to the library

On Linux systems, libraries need to be installed with attached version
information so other packages can depend on specific versions and make
sure they don't compile against a version they're not compatible with

This change installs libs2.so.0.10.0 with appropriate
symlinks as libs2.so.0 and libs2.so.

Do note that for new future versions, the version in CMakeLists.txt will have
to be updated as well
---
 CMakeLists.txt | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a8e87f2..62d411e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,7 @@
 cmake_minimum_required(VERSION 3.12)
 
-project(s2-geometry)
+project(s2-geometry
+	VERSION 0.9.0)
 
 include(CMakeDependentOption)
 include(CheckCXXCompilerFlag)
@@ -214,6 +215,11 @@ target_link_libraries(
 # target_link_libraries(<target_name> s2)
 target_include_directories(s2 PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/src)
 
+# Add version information to the target
+set_target_properties(s2 PROPERTIES
+    SOVERSION ${PROJECT_VERSION_MAJOR}
+    VERSION ${PROJECT_VERSION})
+
 # We don't need to install all headers, only those
 # transitively included by s2 headers we are exporting.
 install(FILES src/s2/_fp_contract_off.h
-- 
2.29.2

